patterns/ecs-v1/java

```diff
@@ -4,14 +4,14 @@ JAVAFILE (?:[A-Za-z0-9_. -]+)
 #Allow special <init>, <clinit> methods
 JAVAMETHOD (?:(<(?:cl)?init>)|[a-zA-Z$_][a-zA-Z$_0-9]*)
 #Line number is optional in special cases 'Native method' or 'Unknown source'
-JAVASTACKTRACEPART %{SPACE}at %{JAVACLASS:class}\.%{JAVAMETHOD:method}\(%{JAVAFILE:file}(?::%{NUMBER:line})?\)
+JAVASTACKTRACEPART %{SPACE}at %{JAVACLASS:[log][origin][class][name]}\.%{JAVAMETHOD:[log][origin][function]}\(%{JAVAFILE:[log][origin][file][name]}(?::%{INT:[log][origin][file][line]})?\)
 # Java Logs
 JAVATHREAD (?:[A-Z]{2}-Processor[\d]+)
 JAVALOGMESSAGE (.*)
 # MMM dd, yyyy HH:mm:ss eg: Jan 9, 2014 7:13:13 AM
 CATALINA_DATESTAMP %{MONTH} %{MONTHDAY}, 20%{YEAR} %{HOUR}:?%{MINUTE}(?::?%{SECOND}) (?:AM|PM)
 # yyyy-MM-dd HH:mm:ss,SSS ZZZ eg: 2014-01-09 17:32:25,527 -0800
 TOMCAT_DATESTAMP 20%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{HOUR}:?%{MINUTE}(?::?%{SECOND}) %{ISO8601_TIMEZONE}
-CATALINALOG %{CATALINA_DATESTAMP:timestamp} %{JAVACLASS:class} %{JAVALOGMESSAGE:logmessage}
+CATALINALOG %{CATALINA_DATESTAMP:timestamp} %{JAVACLASS:[log][origin][class][name]} %{JAVALOGMESSAGE:message}
```

Comment:
as noted (in the PR desc) due lack of convention we're mapping this to `message`
... a likely better alternative would be to use smt like `java.log.message`

Reply:
If this is the main "message" of this event, it should absolutely go to the `message` field at the root of the event 👍  Think of looking at these logs (or any other logs) via the Elastic Logs UI; you wouldn't want the "message" column to be empty, while the message is present elsewhere in the document :-)

Reply:
thank you, my confusion was (started a discussion elsewhere) in what is a 'better' broader (LS) convention: 
whether to first `message => event.original` (and than simply overwrite the `message` field) 
or assume post rewriting `java.log.message => message` 
... first one makes more sense and is more prevalent in patterns already.

I'm adding a note to revisit grok filter's `overwrite => [ message ]` default in ecs compatibility.

Reply:
Don't forget to also rename other instances of `[log][origin][class][name]`, like on this line and on line 17 :-)